### DISCUSSION TITLE
Error message ketika hendak mengubah tipe data menjadi datetime

### DISCUSSION THREAD

**Discussion by MF-Faqih** [teamalgoritma/community#488] [Q&A] [answered]
Selamat pagi, saya sedang mengerjakan latihan P4DA, pada salah satu bagian saya ingin mengubah tipe salah satu data pada dataframe dari object menjadi datetime46 dengan sytax:

`df[df['Release Date'] == df['Release Date'].astype('datetime64')].head()`

dan warning yang muncul:
C:\Users\mffaq\AppData\Local\Temp\ipykernel_8456\1054612513.py:2: FutureWarning: Passing unit-less datetime64 dtype to .astype is deprecated and will raise in a future version. Pass 'datetime64[ns]' instead
  df[df['Release Date'] == df['Release Date'].astype('datetime64')].head()

untuk mengatasi warning seperti ini solunya bagaimana ya? Terimakasih

**Comment by fafilia** [comment, marked as answer]
Dear @MF-Faqih ,

Sedikit memperbaiki dari syntax yang telah Anda gunakan, untuk mengubah sebuah kolom ke tipe data tertentu dapat dilakukan dengan memanggil Series dari dataframe (jika kasusunya kolom yang akan diubah datanya hanya 1 kolom) dengan cara sebagai berikut:

```
dataframe["kolom_name"] = dataframe["kolom_name"].astype("new_data_type")
```
Jadi yang perlu dibenahi dari syntax Anda adalah penggunaan tanda = yang cukup 1 saja dan tidak perlu menggunakan tanda kurung siku berlebih, seperti contoh berikut:
```
df["Relase Date"] = df["Relase Date"].astype("datetime64")
```

Silahkan dicoba terlebih dahulu, semoga penjelasan di atas dapat membantu.

**Reply by MF-Faqih** [reply]
terimakasih kak @fafilia  atas bantuannya, saya mendapatkan hal lain yang menjadi perhatian saya disini. Jika mengeksekusi kode  baru setelah pesan error muncul, output code sudah benar namun pesan error tetap muncul. Pesan error baru akan hilang jika melakukan running dari cell teratas.
